crates/runtime/src/init/dataset.rs

```diff
@@ -197,6 +197,7 @@ impl Runtime {
 
         let source = ds.source();
         let params = ConnectorParamsBuilder::new(source.into(), (&ds).into())
+            .with_app(Arc::clone(&ds.app))
```

Comment:
I don't think it's necessary to pass the app explicitly here - it should be possibel to get app reference from dataset inside `ConnectorParamsBuilder.build` method from dataset component, and pass it down to the connector params.

Reply:
```rs
let (params, prefix, parameters, app) = match &self.component {
    ConnectorComponent::Catalog(catalog) => {
        // ...

        (
            get_params_with_secrets(Arc::clone(&secrets), &catalog.params).await,
            factory.prefix(),
            factory.parameters(),
            None,
        )
    }
    ConnectorComponent::Dataset(dataset) => {
        // ...

        (
            params,
            factory.prefix(),
            factory.parameters(),
            Some(Arc::clone(&dataset.app)),
        )
    }
};
```